services/cd-service/pkg/repository/repository.go

```diff
@@ -523,33 +523,66 @@ func (r *repository) ProcessQueue(ctx context.Context, health *setup.HealthRepor
 	}
 }
 
-func (r *repository) applyTransformerBatches(transformerBatches []transformerBatch, allowFetchAndReset bool, transaction *sql.Tx) ([]transformerBatch, error, *TransformerResult) {
+func (r *repository) applyTransformerBatches(transformerBatches []transformerBatch, allowFetchAndReset bool) ([]transformerBatch, error, *TransformerResult) {
 	//exhaustruct:ignore
 	var changes = &TransformerResult{}
+	transactions := make([]*sql.Tx, len(transformerBatches))
+	absoluteIndex := -1
+	if r.DB.ShouldUseEslTable() {
+		defer func(txs []*sql.Tx) { //Makes sure we always attempt a rollback on every created tx.
+			for _, tx := range txs {
```

Comment:
do we really need this loop?
At any given point in time, there should only be 1 (or 0) transaction open.
I think this would be clearer if it's just one `transaction` not an array.